src/FSharpPlus/Extensions.fs

```diff
@@ -295,6 +295,93 @@ module String =
+    let trim      (trimChars: char seq) (source: string) = source.Trim (Seq.toArray trimChars)
+    /// Removes all leading occurrences of specified characters from the given string.
+    let trimStart (trimChars: char seq) (source: string) = source.TrimStart (Seq.toArray trimChars)
+    /// Removes all trailing occurrences of specified characters from the given string.
+    let trimEnd   (trimChars: char seq) (source: string) = source.TrimEnd (Seq.toArray trimChars)
+
+    let toArray (source: string)    = source.ToCharArray ()
+    let ofArray (source: char [])   = new String (source)
+    let toList  (source: string)    = toArray source |> List.ofArray
+    let ofList  (source: char list) = new String (source |> Array.ofList)
+    let toSeq   (source: string)    = source :> seq<char>
+    let ofSeq   (source: seq<char>) = String.Join (String.Empty, source)
+
+    let item    (index: int) (source: string) = source.[index]
+    let tryItem (index: int) (source: string) = if index >= 0 && index < source.Length then Some source.[index] else None
+
+    let rev (source: string) = new String (source.ToCharArray () |> Array.rev)
+
+    let take count (source: string) = source.[..count-1]
+    let skip count (source: string) = source.[count..]
+    let takeWhile (predicate: char -> bool) (source: string) =
+        if String.IsNullOrEmpty source then
+            String.Empty
+        else
+            let mutable i = 0
+            let length = String.length source
+            while i < length && predicate source.[i] do
+                i <- i + 1
+            if i = 0 then ""
+            else source |> take i
+    let skipWhile (predicate: char -> bool) (source: string) =
+        if String.IsNullOrEmpty source then
+            String.Empty
+        else
+            let mutable i = 0
+            let length = String.length source
+            while i < length && predicate source.[i] do
+                i <- i + 1
+            if i = 0 then ""
+            else source |> skip i
+    /// Returns a string that have at most N characters from the beginning of the original string.
+    /// It returns the original string if it is shorter than count.
+    let truncate count (source: string) =
+        if count < 1 then String.Empty
+        else if String.length source <= count then source
+        else take count source
+    /// Returns a string that drops first N characters of the original string.
+    /// When count exceeds the length of the string it returns an empty string.
+    let drop     count (source: string) =
+        if count < 1 then source
+        else if String.length source >= count then String.Empty
+        else skip count source
+
+    let findIndex    (char: char) (source: string) = source.IndexOf char
+    let tryFindIndex (char: char) (source: string) =
+        let index = findIndex char source
+        if index = -1 then None else Some index
+
+    let findSliceIndex    (slice: string) (source: string) = source.IndexOf slice
```

Comment:
I wonder whether should we throw here, instead of returning `-1`

Reply:
Ah it should throw `ArgumentException` as `List.findIndex` do